### PR TITLE
Ability to deactivate parameter validation in serializer through config

### PR DESCRIPTION
**Pull request by carl0FF**
The natural follow-up to #797 - exposing parameter validation as an option at serialization time when configuring the client!! :)

### CODE REVIEW

File: botocore/client.py [hunk 1]
```diff
@@ -617,6 +618,10 @@ class Config(object):
         thrown when attempting to read from a connection. The default is
         60 seconds.
 
+    :type parameter_validation: str
+    :param parameter_validation: Whether parameter validation should occur
+        when serializing requests. The default is True.
+
```
Comment (carl0FF) [inline botocore/client.py]: My only concern for this PR is whether the caveats so kindly laid out by @JordonPhillips in #796 should be somehow mentioned here or not.

**Comment by jamesls** [comment]
Feature seems reasonable to me.  Any concerns, @kyleknap @JordonPhillips?

**Comment by kyleknap** [comment]
Yeah I am fine with it.

**Comment by JordonPhillips** [comment]
I'm good with it, though I do think the caveats should be mentioned.

**Comment by jamesls** [comment]
Ok, I'll rebase the commits and merge.

**Comment by jamesls** [comment]
Merged.  Thanks for the pull request.

**Pull request closed without merging**